idtoken/idtoken.go

```diff
@@ -50,7 +51,9 @@ func NewClient(ctx context.Context, audience string, opts ...ClientOption) (*htt
 	if err != nil {
 		return nil, err
 	}
-	opts = append(opts, option.WithTokenSource(ts))
+	// skipping validation so added TokenSource will not conflict with user
```

Comment:
```suggestion
	// Skip DialSettings validation so added TokenSource will not conflict with user
```

Also, it looks like `ds.Validate` has already been called above?

Reply:
That one is safe to call, and should be called. The validation only needs to be loosened here because when `WithTokenSource` is added, if the user had supplied other credential type options and future validations would fail.